### ISSUE TITLE
Message "Warning: Your dependencies could not be resolved. You likely have a mismatch in your sub-dependencies." is not always right

### ISSUE THREAD

**Issue by Kroppeb** [pypa/pipenv#6091]
I have encountered the following message a few times:

>  Warning: Your dependencies could not be resolved. You likely have a mismatch in your sub-dependencies.

Some other cases I remember:

* The package being installed doesn't provide the modules it claims it does (we had a typo)
* The package has dependencies in form of `package @ url` where the url is invalid.

The suggestion to use `pipenv run pip install <requirement_name>` does work, as it gives a more usable error. But is a bit useless when you don't know which dependency is causing the issue.

**Comment by matteius** [comment]
I've usually been able to narrow it down by looking at the output of the lock generation when using the `-v` or `--verbose` flag.

**Comment by Kroppeb** [comment]
Yeah, that does help sometimes. Given the fact that pip has better errors, I suppose my question is: is there a way that these could be given to the user somehow?
